File: web/src/api/modules/dict.ts
import service from '@/utils/request'
import type { ApiResponse } from '@/types'

export interface Dict {
  id: number
  dictName: string
  dictCode: string
  description: string
  status: number
  createdAt: string
  updatedAt: string
  itemCount?: number
  dictItems?: DictItem[]
}

export interface DictItem {
  id: number
  dictId: number
  itemKey: string
  itemValue: string
  sort: number
  description: string
  status: number
  createdAt: string
  updatedAt: string
}

/**
 * 获取字典列表（分页）
 * 使用场景：字典管理页面主列表
 */
export const getDictList = (params?: {
  page?: number
  pageSize?: number
  keyword?: string
  name?: string
  code?: string
  status?: number
}): Promise<
  ApiResponse<{
    list: Dict[]
    total: number
    page: number
    pageSize: number
  }>
> => {
  return service({
    url: '/dict',
    method: 'get',
    params,
  })
}

/**
 * 创建字典
 * 使用场景：字典管理页面新增字典
 */
export const createDict = (data: {
  dictName: string
  dictCode: string
  description: string
  status: number
}): Promise<ApiResponse<Dict>> => {
  return service({
    url: '/dict',
    method: 'post',
    data,
  })
}

/**
 * 更新字典
 * 使用场景：字典管理页面编辑字典、切换字典状态
 */
export const updateDict = (
  id: number,
  data: {
    dictName?: string
    dictCode?: string
    description?: string
    status?: number
  },
): Promise<ApiResponse<Dict>> => {
  return service({
    url: `/dict/${id}`,
    method: 'put',
    data,
  })
}

/**
 * 删除字典
 * 使用场景：字典管理页面删除字典（同时删除关联字典项）
 */
export const deleteDict = (id: number): Promise<ApiResponse> => {
  return service({
    url: `/dict/${id}`,
    method: 'delete',
  })
}

/**
 * 创建字典项
 * 使用场景：字典管理页面添加明细项
 */
export const createDictItem = (
  dictId: number,
  data: {
    itemKey: string
    itemValue: string
    sort: number
    description: string
    status: number
  },
): Promise<ApiResponse<DictItem>> => {
  return service({
    url: '/dict/item',
    method: 'post',
    data: { dictId, ...data },
  })
}

/**
 * 更新字典项
 * 使用场景：字典管理页面编辑明细项、切换明细状态
 */
export const updateDictItem = (
  id: number,
  data: {
    dictId?: number
    itemKey?: string
    itemValue?: string
    sort?: number
    description?: string
    status?: number
  },
): Promise<ApiResponse<DictItem>> => {
  return service({
    url: '/dict/item',
    method: 'put',
    data: { id, ...data },
  })
}

/**
 * 删除字典项
 * 使用场景：字典管理页面删除明细项
 */
export const deleteDictItem = (id: number): Promise<ApiResponse> => {
  return service({
    url: `/dict/item/${id}`,
    method: 'delete',
  })
}

/**
 * 根据字典 ID 获取字典项列表
 * 使用场景：字典管理页面打开明细抽屉时加载明细列表
 */
export const getDictItemsByDictId = (
  dictId: number,
): Promise<ApiResponse<{ list: DictItem[]; total: number }>> => {
  return service({
    url: '/dict/items',
    method: 'get',
    params: { dict_id: dictId },
  })
}

/**
 * 根据字典编码获取启用的字典项
 * 使用场景：全局下拉框、表单选项等需要字典选项的场景
 */
export const getDictItemsByCode = (code: string): Promise<ApiResponse<DictItem[]>> => {
  return service({
    url: `/dict/code/${code}/active-items`,
    method: 'get',
  })
}

File: server/internal/dto/base/response/dict.go
package response

import "github.com/ix-pay/ixpay-pro/internal/utils/common/baseRes"

// DictItemResponse 字典项响应模型
// 所有 ID 字段使用 int64 格式，通过 json:",string" 标签自动序列化为字符串
type DictItemResponse struct {
	ID          int64  `json:"id,string"`     // 字典项 ID
	DictID      int64  `json:"dictId,string"` // 字典 ID
	ItemKey     string `json:"itemKey"`       // 字典项键
	ItemValue   string `json:"itemValue"`     // 字典项值
	Sort        int    `json:"sort"`          // 排序
	Description string `json:"description"`   // 描述
	Status      int    `json:"status"`        // 状态：1-启用 0-禁用
	CreatedAt   string `json:"createdAt"`     // 创建时间
	UpdatedAt   string `json:"updatedAt"`     // 更新时间
}

// DictResponse 字典响应模型
type DictResponse struct {
	ID          int64              `json:"id,string"`   // 字典 ID
	DictName    string             `json:"dictName"`    // 字典名称
	DictCode    string             `json:"dictCode"`    // 字典编码
	Description string             `json:"description"` // 描述
	Status      int                `json:"status"`      // 状态：1-启用 0-禁用
	CreatedAt   string             `json:"createdAt"`   // 创建时间
	UpdatedAt   string             `json:"updatedAt"`   // 更新时间
	DictItems   []DictItemResponse `json:"dictItems"`   // 字典项列表
}

// DictListResponse 字典列表响应模型
type DictListResponse struct {
	baseRes.PageResult
	List []DictResponse `json:"list"` // 字典列表
}

// DictItemListResponse 字典项列表响应模型
type DictItemListResponse struct {
	baseRes.PageResult
	List []DictItemResponse `json:"list"` // 字典项列表
}
